pacfetcher.go

```diff
@@ -80,6 +83,46 @@ func requireOK(resp *http.Response, err error) (*http.Response, error) {
 	}
 }
 
+// decodeDataURL decodes a data URL (e.g., data:text/plain;base64,SGVsbG8sIFdvcmxkIQ==).
+// It supports both base64 and URL-encoded data, and enforces the maxResponseBytes size limit.
+// See https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs for details.
+func decodeDataURL(uri string) ([]byte, error) {
+	parsed_url, err := url.Parse(uri)
+
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing pac url: %w", err)
+	}
+
+	if parsed_url.Scheme != "data" {
+		return nil, nil
+	}
+	parts := strings.SplitN(parsed_url.Opaque, ",", 2)
+	if len(parts) < 2 {
+		return nil, fmt.Errorf("Error parsing data URL: Invalid Format")
+	}
+
+	isBase64 := slices.Contains(strings.Split(parts[0], ";"), "base64")
+	if isBase64 {
+		bytes, err := base64.StdEncoding.DecodeString(parts[1])
+		if err != nil {
+			return nil, fmt.Errorf("Error parsing base64 data URL: %w", err)
+		}
+		if len(bytes) > maxResponseBytes {
+			return nil, fmt.Errorf("Error parsing base64 data URL: PAC JS is too big (limit is %d bytes)", maxResponseBytes)
+		}
+		return bytes, nil
+	}
+
+	decoded, err := url.QueryUnescape(parts[1])
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing data URL: %w", err)
+	}
+	if len(decoded) > maxResponseBytes {
+		return nil, fmt.Errorf("Error parsing data URL: PAC JS is too big (limit is %d bytes)", maxResponseBytes)
+	}
```

Comment:
i'd suggest doing this check once, before we try do decode the data: url. we should probably use a different limit than `maxResponseBytes`. from your mdn link:

> Browsers are not required to support any particular maximum length of data. Chromium and Firefox limit data URLs to 512MB, and Safari (WebKit) limits them to 2048MB. Note that Firefox 97 increased the limit from 256KB to 32MB, and [Firefox 136 increased it to 512MB](https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Releases/136).